arch/arc/core/fast_irq.S

```diff
@@ -238,7 +238,7 @@ _firq_reschedule:
 	pop_s r2
 #endif
 
-	ld_s r3, [r2, _thread_offset_to_relinquish_cause]
+	ld r3, [r2, _thread_offset_to_relinquish_cause]
```

Comment:
Just curious, why ld and not the ld_s? Afraid the _thread_offset_ might by too large for the 16-bit instruction encoding of the ld_s variant?

Reply:
@ruuddw yes, with this config enabled I get compilation errors because the offset is too large, so I made this change

Reply:
Thanks, I missed the commit message. Thread struct can become large with this static name buffers, but I didn't expect it to overflow already. Anyway, the few bytes saved by the ld_s instruction don't make that much of a difference on the total.